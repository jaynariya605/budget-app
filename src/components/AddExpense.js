import React from "react";
import ExpenseForm from "./ExpenseForm";
import { connect } from "react-redux";
import { startAddExpense } from "../actions/expenses";
import {  useNavigate } from "react-router-dom";


export const AddExpense =(props)=>{
    const navigate = useNavigate()
    const onSubmit = (expense) =>{
       props.startAddExpense(expense)
        navigate("/dashboard")
    }

    
        return (
      
                <div>
                    <div className="page-header">
                        <div className="content-container">
                        <h1 className="page-header__title">Add Expense</h1>
                        </div>
                    </div>
                    <div className="content-container">
                        <ExpenseForm 
                        onSubmit = {onSubmit}
                        />
                    </div>
                    
                </div>
                
           
            
        )
    

}


const mapDispatchToprops = (dispatch) => ({
    startAddExpense: (expense) => dispatch(startAddExpense(expense)),
    })


export default connect(undefined,mapDispatchToprops )(AddExpense)