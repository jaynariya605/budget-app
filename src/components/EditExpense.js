import React from "react";
import { connect } from "react-redux";
import { useParams } from "react-router-dom";
import ExpenseForm from "./ExpenseForm";
import { startEditExpense, startRemoveExpense } from "../actions/expenses";
import { useNavigate  } from "react-router-dom";


export const EditExpense =(props)=>{
    const navigate = useNavigate()
    const onSubmit = (update)=>{
        props.startEditExpense(props.match.id, update)
        navigate("/dashboard")
    }

    const onClick = ()=> {
        props.startRemoveExpense({ id : props.match.id })
       navigate("/dashboard")
    }

  
    const expense = props.expenses.find((expense) => expense.id === props.match.id)

        return (
            <div>
                <div className="page-header">
                    <div className="content-container">
                        <h1 className="page-header__title">Edit Expense</h1>
                    </div>
                </div>
                <div className="content-container">
                    <ExpenseForm 
                    expense={expense}
                    onSubmit={onSubmit}/>
                    <button className="button button--secondary" onClick={onClick}>Remove</button>
                </div>
            </div>
                
           
    )
    

}

const Wrapper = (props) => {
    let params = useParams()
    return (<EditExpense { ...props } match={params} />)
}



const mapStateToProps = (state) =>{
    return {
        expenses: state.expenses, 
    }
}

const mapDispatchToprops = (dispatch) =>{
    return {
        startEditExpense: (id, update) => dispatch(startEditExpense(id, update)),
        startRemoveExpense: ({id}) => dispatch(startRemoveExpense({id}))
    }
}

export  default connect(mapStateToProps, mapDispatchToprops)(Wrapper)