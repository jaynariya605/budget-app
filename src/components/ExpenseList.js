import React, { Suspense, useEffect, useState } from "react";
import { connect } from "react-redux";
import ExpenseListItem from './ExpenseListItem';
import selectExpenses from '../selectors/expenses';



export const ExpenseList = (props) =>{
    
    const [data, setData] = useState([])
    const [isLoaded, setLoading] = useState(false)
    const [isFirst,setFirst] = useState(true)

    useEffect(()=>{
        setTimeout(()=>{ 
            setLoading(true)
            setFirst(false)
        }, 5000)
        setData(props.expenses)
        
        
    },[props.expenses])
        
    
    
    return (
    <div className="content-container">
        <div className="list-header">
            <div className="show-for-mobile">Expenses</div>
            <div className="show-for-desktop">Expense</div>
            <div className="show-for-desktop">Amount</div>
        </div>

        
        <div className="list-body">
        
        

        {((!isFirst && data.length ===0) ? (
            <div className="list-item__message">
                <span>No expense added</span>
            </div>
            ) : 
        
        (
            
            data.map((expense)=>{
                return (
                    <div key={expense.id}>
                    
                        <ExpenseListItem  { ...expense}/>
                    
                    
                    </div>
                )
            })
            
        ))
    }
    {!isLoaded && <div className="loadinglist"> <img className="loader__image--list" src="/img/loader.gif"></img></div>}
    
    
    </div>
    </div>
)}

const mapStateToProps = (state)=>{
    return  {
        expenses: selectExpenses(state.expenses,state.filters),
    }
}

export default connect(mapStateToProps)(ExpenseList);

