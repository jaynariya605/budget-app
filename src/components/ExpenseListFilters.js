import React from "react";
import { connect } from "react-redux";
import { setTextFilter, sortByDate, sortByAmount, setStartDate, setEndDate } from "../actions/filters";
//import 'react-dates-2/initialize';
import { DateRangePicker } from 'react-dates-2';
import 'react-dates-2/lib/css/_datepicker.css';


export class ExpenseListFilters extends React.Component{

    state = {
        calanderFocused: null,
    }

    onDatesChange = ({ startDate, endDate }) => {
        this.props.setStartDate(startDate)
        this.props.setEndDate(endDate)
    }

    onFocusChange = (calanderFocused) => {
        this.setState(()=> ({ calanderFocused }))
    }
    
    onTextChange = (e)=>{
        this.props.onTextChange(e.target.value)
    }

    onSortChange = (e)=>{
        const select = e.target.value === 'date' ? sortByDate : sortByAmount
        this.props.onSortChange(select)
    }
    render () {
        return (
            <div className="content-container">
                <div className="input-group">
                    <div className="input-group__item">
                    <input placeholder="Search Expenses" className="text-input" type="text" defaultValue={this.props.filters.text} onChange={this.onTextChange} />
                    </div>
                    <div className="input-group__item">
                        <select className="select" defaultValue={this.props.filters.sortBy} onChange={this.onSortChange}>
                            <option value="date">Date</option>
                            <option value="amount">Amount</option>
                        </select>
                    </div>
                    <div className="input-group__item">
                        
                        <DateRangePicker
                        startDateId="MyDatePicker"
                        startDate={this.props.filters.startDate}
                        endDateId="MyDatePicker"
                        endDate={this.props.filters.endDate}
                        onDatesChange={this.onDatesChange}
                        focusedInput= {this.state.calanderFocused}
                        onFocusChange={this.onFocusChange}
                        showClearDates={true}
                        numberOfMonths={1}
                        isOutsideRange={()=>false}
                        />
                            
                    </div>
                </div>
                

            </div>
        )
    }
}



const mapStateToProps = (state) => {
    return {
        filters: state.filters
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
        onSortChange: (select) => dispatch(select()),
        onTextChange: (text)=>dispatch(setTextFilter(text)),
        setStartDate: (startDate) => dispatch(setStartDate(startDate)),
        setEndDate: (endDate) => dispatch(setEndDate(endDate))
    }
}
export default connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilters)