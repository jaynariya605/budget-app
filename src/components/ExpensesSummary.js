import React from "react";
import { connect } from "react-redux";
import getVisibleExpenses from"../selectors/expenses";
import getTotalExpenses from"../selectors/expenses-total"
import numeral from "numeral";
import  { Link } from 'react-router-dom'

export const ExpenseSummary= ({ expenseCount, expensesTotal }) =>{
    const formatedTotal = numeral(expensesTotal / 100).format('$0,0.00')
    return (
        <div className="page-header">
            <div className="content-container">
                <h1 className="page-header__title"> There are total <span>{expenseCount}</span> expenses of <span>{formatedTotal}</span> . </h1>
                <div className="page-header__actions">
                    <Link className="button" to="/create">Add Expense</Link>
                </div>
                
            </div>
            
        </div>
    )
}

const MapstateToProps = (state) =>{
    const visibleExpenses =  getVisibleExpenses(state.expenses, state.filters)
    return {
        expenseCount: visibleExpenses.length,
        expensesTotal: getTotalExpenses(visibleExpenses)
    }
}

export default connect(MapstateToProps)(ExpenseSummary)