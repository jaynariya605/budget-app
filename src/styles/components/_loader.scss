.loader {
    height: 100vh;
    width: 100vw;
    display: flex;
    justify-content: center;
    align-items: center;
}

.loader__image {
    height: 6rem;
    width: 6rem;
}

.loader__image--list {
    height: 3rem;
    width: 3rem;
}

.loadinglist {
    display: flex;
    align-items: center;
    justify-content: center;
    margin-top: 16px;
}