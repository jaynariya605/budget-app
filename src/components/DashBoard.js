import React from "react";
import ExpenseList from './ExpenseList';
import ExpenseListFilters from "./ExpenseListFilters";
import ExpensesSummary from "./ExpensesSummary";
import { useEffect } from "react";


export const DashBoard =  () => {
            return (<div>
            <ExpensesSummary/>
            <ExpenseListFilters/>
            <ExpenseList />
            </div>)
      

};


export default DashBoard

