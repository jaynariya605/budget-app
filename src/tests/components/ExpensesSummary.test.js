import { ExpenseSummary } from "../../components/ExpensesSummary";
import React from "react";
import { shallow } from "enzyme";

test('should render correctl ExepenssSummary ', ()=>{
    const wrapper = shallow(<ExpenseSummary expenseCount={1} expensesTotal={235} />)
    expect(wrapper).toMatchSnapshot()
})