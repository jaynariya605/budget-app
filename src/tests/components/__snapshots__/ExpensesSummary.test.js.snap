// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctl ExepenssSummary  1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
       There are total 
      <span>
        1
      </span>
       expenses of 
      <span>
        $2.35
      </span>
       . 
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;
